src/watcher.ts

```diff
@@ -321,7 +321,11 @@ module.exports = class MonoplasmaWatcher extends EventEmitter {
                 return block.timestamp * 1000
             })()
         }
-        return await this.blockTimestampCache[blockNumber]
+        if (this.blockTimestampCache[blockNumber].constructor.name === "Promise") {
```

Comment:
I think the canonical way to do this is to ducktype on `then`: `if (typeof this.blockTimestampCache[blockNumber].then === 'function')` Not that we're using them anywhere but it allows this code to also work for promise subclasses/wrappers and thenables.